src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala

```diff
@@ -64,10 +62,22 @@ class AgoraMongoDao(collection: MongoCollection) extends AgoraDao {
   }
 
   def find(query: Imports.DBObject) = {
-    (for (entity <- collection.find(query)) yield grater[AgoraEntity].asObject(entity)).toVector
+    (for (dbObject <- collection.find(query)) yield MongoDbObjectToEntity(dbObject)).toVector
+  }
+
+  override def find(entity: AgoraEntity): Seq[AgoraEntity] = find(EntityToMongoDbObject(entity))
+
+  override def findSingle(entity: AgoraEntity): AgoraEntity = {
+    val entityVector = find(EntityToMongoDbObject(entity))
+    entityVector.length match {
+      case 1 => entityVector.head
+      case 0 => throw new Exception("Found 0 documents matching: " + entity.toString)
```

Comment:
Add logger.error writes alongside these exceptions, now that we have logging :)

Should we go ahead and make our own exception class?